Initial support for generating map previews

Crotalus opened the pull request:
- Extract DDS stream from maps
- DDS -> PNG previews

As mentioned in comment / github issue Wand didn't play well with 32-bit environment.

Feel free to use / steal parts from this for the map upload api

Crotalus committed:
Initial support for generating map previews

Sheeo commented:
What 32-bit environment did you try running it in? When I was running it it was working fine.

micheljung commented:
Is there really no solution in Python that does not require a tool like imagemagick to be installed on the system?

Update: OK I see it's an optional alternative

Crotalus commented:
Yeah, experimented with some different solutions. Started with Wand but it crashed for me so then IM-convert and finally PIL.

I will provide info about the Wand crash to sheeo, although it's midsummer celebration here atm so I will probably be mentally passed out for a couple of days.

micheljung committed:
Generate map previews incl. markers
And clean up utility so it assumes less about where the maps are put, how they are named, what sizes they are etc.
